src/lib/eina/eina_inline_lock_win32.x

```diff
@@ -209,31 +185,25 @@ _eina_lock_take_try(Eina_Lock *mutex)
         LeaveCriticalSection(_eina_tracking_lock);
      }
 #endif
+
    return ret;
 }
 
 static inline Eina_Lock_Result
 _eina_lock_take(Eina_Lock *mutex)
 {
-   Eina_Lock_Result ret = EINA_LOCK_FAIL;
-   DWORD ok;
-
-   SetLastError (ERROR_SUCCESS);
-
 #ifdef EINA_HAVE_ON_OFF_THREADS
    if (!_eina_threads_activated) return EINA_LOCK_SUCCEED;
 #endif
 
 #ifdef EINA_HAVE_DEBUG_THREADS
-   if (eina_lock_take_try(mutex) == EINA_LOCK_SUCCEED) return EINA_LOCK_SUCCEED;
-
    if (_eina_threads_debug >= 100)
      {
         struct timeval t0, t1;
         int dt;
 
         gettimeofday(&t0, NULL);
-        ok = EnterCriticalSection(&(mutex->mutex));
+        EnterCriticalSection(mutex->mutex);
```

Comment:
`EnterCriticalSection` receives `LPCRITICAL_SECTION` as arg, so shouldn't it be like:
```suggestion
        EnterCriticalSection(&(mutex->mutex));
```

Reply:
Good catch, thanks!